scripts/functions/vae.py

```diff
@@ -0,0 +1,329 @@
+# -----------------------------------------------------------------------------------------------------------------------
+# By Alexandra Lee 
+# (updated October 2018) 
+#
+# Encode gene expression data into low dimensional latent space using 
+# Tybalt with 2-hidden layers
+# --------------------------------------------------------------------------------------------------------------------
+import os
+import argparse
+import pandas as pd
+import matplotlib.pyplot as plt
+import tensorflow as tf
+
+# To ensure reproducibility using Keras during development
+# https://keras.io/getting-started/faq/#how-can-i-obtain-reproducible-results-using-keras-during-development
+import numpy as np
+import tensorflow as tf
+import random as rn
+
+# The below is necessary in Python 3.2.3 onwards to
+# have reproducible behavior for certain hash-based operations.
+# See these references for further details:
+# https://docs.python.org/3.4/using/cmdline.html#envvar-PYTHONHASHSEED
+# https://github.com/keras-team/keras/issues/2280#issuecomment-306959926
+randomState = 123
+import os
+os.environ['PYTHONHASHSEED'] = '0'
+
+# The below is necessary for starting Numpy generated random numbers
+# in a well-defined initial state.
+
+np.random.seed(42)
+
+# The below is necessary for starting core Python generated random numbers
+# in a well-defined state.
+
+rn.seed(12345)
+
+# Force TensorFlow to use single thread.
+# Multiple threads are a potential source of
+# non-reproducible results.
+# For further details, see: https://stackoverflow.com/questions/42022950/which-seeds-have-to-be-set-where-to-realize-100-reproducibility-of-training-res
+
+session_conf = tf.ConfigProto(intra_op_parallelism_threads=1, inter_op_parallelism_threads=1)
+
+from keras import backend as K
+
+# The below tf.set_random_seed() will make random number generation
+# in the TensorFlow backend have a well-defined initial state.
+# For further details, see: https://www.tensorflow.org/api_docs/python/tf/set_random_seed
+
+tf.set_random_seed(1234)
+
+sess = tf.Session(graph=tf.get_default_graph(), config=session_conf)
+K.set_session(sess)
+
+
+from keras.layers import Input, Dense, Lambda, Layer, Activation
+from keras.layers.normalization import BatchNormalization
+from keras.models import Model, Sequential
+from keras import metrics, optimizers
+from keras.callbacks import Callback
+
+def tybalt_2layer_model(learning_rate, batch_size, epochs, kappa, intermediate_dim, latent_dim, epsilon_std, base_dir, analysis_name):
```

Comment:
may be worth checking out [`tybalt.data_models`](https://github.com/greenelab/tybalt/blob/master/tybalt/data_models.py#L179).

Can use like `from tybalt import DataModel`. I don't have a 2 layer mode here, but contributions are welcome!

Reply:
This is definitely a good idea and something started to think about during this refactor, however I think this change will need to wait in order to get additional analysis completed.  Hopefully in the next refactor.